packages/angular_devkit/build_angular/src/differential-loading/utils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as browserslist from 'browserslist';
+import * as caniuse from 'caniuse-api';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+
+export function isDifferentialLoadingNeeded(
+  tsConfigFileName: string, browserslistQuery: string[] | undefined = undefined): boolean {
+  const es5SupportNeeded = isEs5BrowserSupportNeeded(tsConfigFileName, browserslistQuery);
+  const target = getTargetVersion(tsConfigFileName);
+
+  return es5SupportNeeded && target !== ts.ScriptTarget.ES5;
+}
+
+export function getTargetVersion(tsConfigFileName: string): ts.ScriptTarget {
```

Comment:
is it possible to reference a `ts.Program` already at this point, so we don't have to worry about things like whether we found the right `tsconfig` file?

Reply:
We're using this to get the `target` output. We determine the maximum version (i.e. the `script[module]` one) from this property.

Reply:
sure, but how does the user of this API know how to locate the correct tsconfig file if there are several in the project?

Reply:
The user of this API will be the browser builder which finds the tsconfig.json to use in the `angular.json`. It can also get overruled by a command line parameter.

Reply:
If the user of this is browser builder, we already know the script target and if es5Support is needed or not. As there is where we modify the mainFields to include es2015 or not. So why do we need to get this again?

When you say that it can be overrules by a command line parameter, can you please elaborate? Wouldn't such flag be in the browser builder?

I think in general there is a bit of a lack of understanding from me on how everything will be glued together.

Reply:
@alan-agius4 what you're saying sounds right and aligns with the design. On `browserlist` depends if we should generate ES5 bundles or not.

Reply:
Ok, I see what you mean. THX!